### PR TITLE
--list-test-names-and-sources

### PR DESCRIPTION
**Pull request by Cyberboss**
I'm working on a Visual Studio unit test adapter for Catch and was looking for a way to view the source file and line of a test without actually running it. So I implemented this

Sample output:

Scenario: Test that threaded operations can be canceled
s:\documents\actual documents\da git\cyberenginemkiii\tests\platform\system\threads.cpp(78)
Scenario: Test basic sleep works
s:\documents\actual documents\da git\cyberenginemkiii\tests\platform\system\threads.cpp(109)
Scenario: Test basic yield works
s:\documents\actual documents\da git\cyberenginemkiii\tests\platform\system\threads.cpp(126)
Scenario: Test thread system errors work
s:\documents\actual documents\da git\cyberenginemkiii\tests\platform\system\threads.cpp(150)

### CODE REVIEW

**Comment by cybnetsurfe3011** [comment]
Clang builds failed because http://llvm.org/apt/llvm-snapshot.gpg.key 404'd

**Comment by nabijaczleweli** [comment]
Can confirm, `llvm.org/apt` has been down for the last couple of days

File: include/internal/catch_config.hpp [hunk 1]
```diff
@@ -50,7 +52,9 @@ namespace Catch {
         bool listTests;
         bool listTags;
         bool listReporters;
+        bool listTestNamesAndSources;
         bool listTestNamesOnly;
+		bool listTestSources;
```
Comment (martinmoene) [inline include/internal/catch_config.hpp]: Mmm, a Tab character!

Comment (cybnetsurfe3011) [inline include/internal/catch_config.hpp]: A) VS is a sneaky mistress
B) Whoops, I've been commiting to master when I have a pull request open

**Comment by philsquared** [comment]
Thanks for this (and apologies for the delay in getting to it). While this looks like a useful PR I'm about to make some changes elsewhere that will make it redundant.
That is, I'm going to make `-l` pass off to the reporter to format its output. That way you can use the xml reporter to provide machine readable output that will have this information.

**Comment by Cyberboss** [comment]
Alright, no problem

**Pull request closed without merging**